src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -15,95 +15,99 @@
     {
+        uint8_t buffer[kMaxSerializedSize] = { 0 };
+        DefaultStorageKeyAllocator key;
+        // Update storage key
+        UpdateKey(key);
+
         // Serialize the data
         TLV::TLVWriter writer;
-        writer.Init(mBuffer, sizeof(mBuffer));
+        writer.Init(buffer, sizeof(buffer));
         ReturnErrorOnFailure(Serialize(writer));
 
-        // Update storage key
-        UpdateKey();
         // Save serialized data
-        ReturnErrorOnFailure(storage.SyncSetKeyValue(mKey.Value(), mBuffer, static_cast<uint16_t>(writer.GetLengthWritten())));
+        ReturnErrorOnFailure(storage.SyncSetKeyValue(key.KeyName(), buffer, static_cast<uint16_t>(writer.GetLengthWritten())));
 
         // WARNING: This read-back is a walkaround for a bug in the Darwing implementation of KeyValueStoreMgr
         //          See https://github.com/project-chip/connectedhomeip/issues/12174.
-        uint16_t size = static_cast<uint16_t>(sizeof(mBuffer));
-        ReturnErrorOnFailure(storage.SyncGetKeyValue(mKey.Value(), mBuffer, size));
-
-        return CHIP_NO_ERROR;
+        uint16_t size = static_cast<uint16_t>(sizeof(buffer));
+        return storage.SyncGetKeyValue(key.KeyName(), buffer, size);
     }
 
     CHIP_ERROR Load(chip::PersistentStorageDelegate & storage)
     {
+        uint8_t buffer[kMaxSerializedSize] = { 0 };
+        DefaultStorageKeyAllocator key;
+
+        // Set data to defaults
+        Clear();
+
         // Update storage key
-        UpdateKey();
+        UpdateKey(key);
+
         // Load the serialized data
-        uint16_t size = static_cast<uint16_t>(sizeof(mBuffer));
-        ReturnErrorOnFailure(storage.SyncGetKeyValue(mKey.Value(), mBuffer, size));
+        uint16_t size = static_cast<uint16_t>(sizeof(buffer));
+        ReturnErrorOnFailure(storage.SyncGetKeyValue(key.KeyName(), buffer, size));
 
         // Decode serialized data
-        Clear();
         TLV::TLVReader reader;
-        reader.Init(mBuffer, size);
-        ReturnErrorOnFailure(Deserialize(reader));
+        reader.Init(buffer, size);
 
-        return CHIP_NO_ERROR;
+        return Deserialize(reader);
     }
 
     CHIP_ERROR Delete(chip::PersistentStorageDelegate & storage)
     {
+        DefaultStorageKeyAllocator key;
+
         // Update storage key
-        UpdateKey();
+        UpdateKey(key);
         // Delete stored data
-        return storage.SyncDeleteKeyValue(mKey.Value());
+        return storage.SyncDeleteKeyValue(key.KeyName());
     }
-
-protected:
-    PersistentStorageKey mKey;
-    uint8_t mBuffer[S] = { 0 };
-    char mDebug[512];
 };
 
 struct FabricData : public PersistentData<kPersistentBufferMax>
 {
-    static const TLV::Tag kFirstEndpoint = 1;
-    static const TLV::Tag kEndpointCount = 2;
-    static const TLV::Tag kFirstKeyset   = 3;
-    static const TLV::Tag kKeysetCount   = 4;
+    static const TLV::Tag kTagFirstEndpoint = TLV::ContextTag(1); // TLV::ContextTag(1); --> CHIP_ERROR_INVALID_TLV_TAG
```

Comment:
What is the comment `// TLV::ContextTag(1); --> CHIP_ERROR_INVALID_TLV_TAG`

Reply:
Sorry for that. It was a note to myself. Removed.